packages/portalnetwork/src/wire/utp/PortalNetworkUtp/index.ts

```diff
@@ -33,17 +40,27 @@ export interface INewRequest {
   contents?: Uint8Array[]
 }
 export class PortalNetworkUTP extends BasicUtp {
-  openContentRequest: Record<UtpSocketKey, ContentRequest> // TODO enable other networks
+  openContentRequest: Map<UtpSocketKey, ContentRequest> // TODO enable other networks
   logger: Debugger
   working: boolean
 
   constructor(logger: Debugger) {
     super()
     this.logger = logger.extend(`uTP`)
-    this.openContentRequest = {}
+    this.openContentRequest = new Map()
     this.working = false
   }
 
+  closeRequest(nodeId: string) {
+    this.logger.extend('CLOSING')(`Closing uTP request with ${nodeId} due to failed connection`)
+    const key = Object.keys(this.openContentRequest).find(
+      (k) => k.slice(0, 5) === nodeId.slice(0, 5)
+    )
+    !key && this.logger.extend('CLOSING')('Cannot find request')
```

Comment:
This is a small thing but can we make this something like below?  Otherwise, we're doing multiple comparisons when really only one is needed (whether or not `key` is defined or not)
```js
if (key === undefined) {
    this.logger.extend...
}
else {
    this.openContentRequest.get...
    this.openContentRequest...
}
```

Reply:
I was trying to find a way for PortalNetwork to signal to uTP to close a request when the discv5 connection to the peer has been interrupted, using just the PeerId.  
```
      try {
        await this.sendPortalNetworkMessage(enr ?? peerId, msg, protocolId, true)
      } catch {
        this.uTP.closeRequest(peerId)
      }
```
Looking at it again, though, I could just deserialize `msg` and get the connectionIds, and the added search.